nni/compression/pytorch/speedup/compress_modules.py

```diff
@@ -152,24 +163,120 @@ def replace_conv2d(conv, mask):
         for groupid in range(conv.groups):
             start = groupid * input_step
             end = (groupid + 1) * input_step
-            current_input_index = list(filter(lambda x: start <= x and x < end, in_channels_index.tolist()))
+            current_input_index = list(
+                filter(lambda x: start <= x and x < end, in_channels_index.tolist()))
             if not current_input_index:
                 # there is no kept channel in current group
-                continue
+                # TODO bug here, the groups is directly get from conv.groups, if the whole group is removed,
+                # then the number of groups in the new_conv also need to change
+                raise Exception(
+                    " Donnot support removing the whole group filter except in the depth-wise conv temporarily")
             # shift the global index into the group index
             current_input_index = [x-start for x in current_input_index]
             # if the groups is larger than 1, the input channels of each
             # group should be pruned evenly.
             assert len(current_input_index) == in_channels_group, \
                 'Input channels of each group are not pruned evenly'
-            current_input_index = torch.tensor(current_input_index).to(tmp_weight_data.device) # pylint: disable=not-callable
+            current_input_index = torch.tensor(current_input_index).to(tmp_weight_data.device)  # pylint: disable=not-callable
             f_start = groupid * filter_step
             f_end = (groupid + 1) * filter_step
-            new_conv.weight.data[f_start:f_end] = torch.index_select(tmp_weight_data[f_start:f_end], 1, current_input_index)
+            new_conv.weight.data[f_start:f_end] = torch.index_select(
+                tmp_weight_data[f_start:f_end], 1, current_input_index)
     else:
         new_conv.weight.data.copy_(tmp_weight_data)
 
     if conv.bias is not None:
-        new_conv.bias.data.copy_(conv.bias.data if tmp_bias_data is None else tmp_bias_data)
+        new_conv.bias.data.copy_(
+            conv.bias.data if tmp_bias_data is None else tmp_bias_data)
 
     return new_conv
+
+
+def replace_convtranspose2d(convtrans, mask):
```

Comment:
When mask is None, may be just return convtrans will be simpler?

Reply:
you mean add the following code snippet at the start?
```
if mask is None:
   return convtrans
```

Reply:
Yeah. If both of mask.input_mask and mask.output_mask are None, we will still create a new module and copy related parameters but it is unnecessary. But I am not sure whether this situation exists or not.

Reply:
https://github.com/microsoft/nni/blob/622e3331c946defe3be529ee7584d3f713f4c70f/nni/compression/pytorch/speedup/compressor.py#L151 
We only replace the modules that have inferred masks.